Automattic/jetpack
# Don't Rely on HTTP Requests for Shortcode Tests

#### Changes proposed in this Pull Request:
1.  Use fake, predetermined HTTP responses via the `pre_http_request` filter for:
   * Ted
   * Instagram
   * Tweet
2. Standardize logic in tests about when to allow HTTP requests (if `@group external-http`), and when to bypass them (otherwise).

#### Testing instructions:
1. `yarn docker:phpunit --filter=WP_Test_Jetpack_Shortcodes`.
2. See that all tests pass and most are quick.
3. `yarn docker:phpunit --filter=WP_Test_Jetpack_Shortcodes --group=external-http`
4. See that all tests pass and that most are slow.

#### Proposed changelog entry for your changes:
None needed
